### ISSUE TITLE
Importing issue

### ISSUE THREAD

**Issue by jbanyai** [partkeepr/PartKeepr#883]
# Error description
I'm trying to import a Part list from CSV with stock levels with the settings can be seen on the following screenshots. 

![image](https://user-images.githubusercontent.com/11853521/28921339-d7cb4b6e-7855-11e7-9816-ce1b5cdaa476.png)

![image](https://user-images.githubusercontent.com/11853521/28921348-e4b8c586-7855-11e7-9ae1-5de5afc34f4e.png)

![image](https://user-images.githubusercontent.com/11853521/28921358-f21d9076-7855-11e7-978e-bf96c167a505.png)

![image](https://user-images.githubusercontent.com/11853521/28921370-01697cfc-7856-11e7-8dda-31a75a7d3f4d.png)

![image](https://user-images.githubusercontent.com/11853521/28921387-09940d66-7856-11e7-8e38-80a07abdb388.png)

When I click on the Export button, the following error appears in the log:
`[2017-08-03 14:04:23] request.CRITICAL: Uncaught PHP Exception Exception: "The key importBehaviour does not exist!" at /var/www/html/PartKeepr/src/PartKeepr/ImportBundle/Configuration/EntityConfiguration.php line 31 {"exception":"[object] (Exception(code: 0): The key importBehaviour does not exist! at /var/www/html/PartKeepr/src/PartKeepr/ImportBundle/Configuration/EntityConfiguration.php:31)"} []
`

I start importing operation on Root Category
If I try to leave the stock information import, the error is the same. 

# System Information
![image](https://user-images.githubusercontent.com/11853521/28921209-49051b08-7855-11e7-9be6-9cec04225f66.png)

**Comment by djklkk** [comment]
Have the same issue on RedHat 6.9 running php5.6.  The error message in partkeepr.log
is about the same:
request.CRITICAL: Uncaught PHP Exception Exception: "The key importBehaviour does not exist for path /!" at /data/www/partkeepr-1.2.0/src/PartKeepr/ImportBundle/Configuration/EntityConfiguration.php line 31 {"exception":"[object] (Exception(code: 0): The key importBehaviour does not exist for path /! at /data/www/partkeepr-1.2.0/src/PartKeepr/ImportBundle/Configuration/EntityConfiguration.php:31)"} []

**Comment by felicitus** [comment]
There were some changes regarding this functionality yesterday, can you try on the demo system if it works for you now?

**Comment by raklet** [comment]
I am having this issue as well.  I have  also tried to import parts on the demo system and it does not function.  

Aside from this error message, the import dialogue also does not let me check any fields for import other than the two fields that are checked by default (name, category).  I can enter other fields and setup a mapping to the csv file (csv will upload and display its information), but I cannot check mark them.  The csv file shows it's information on the first tab, but the preview tab shows the error about "key importBehaviour".  Pressing the import button does not appear to do anything.

**Comment by raklet** [comment]
I did some playing around and discovered that I must click on "create blank preset" first.  As long as I do that the import will work.  It does not matter which configuration I choose in that dialogue - they all work.

**Comment by djklkk** [comment]
Following your lead I did the same and my worked as well

**Comment by christianlupus** [comment]
@jbanyai @djklkk @raklet Can you please check if the current demo page does have the issue? If no, could one of you please document the steps involved (ideally with screenshots) so that we might put it in the wiki? Thank you.

Otherwise, I assume the problem to be fixed.

**Comment by ARNBender** [comment]
We can confirm that this remains an issue using:

Git Revision: 426f6428

And PHP: 7.4.3
Doctrine DBAL 2.5.2
Doctrine ORM 2.5.4

Running on archlinux and nginx

**Comment by christianlupus** [comment]
@ARNBender thank you for checking this for us.

Nevertheless, I am asking myself: How did you get PartKeepr run under 7.4? Did you patch things somehow? There are quite some issues here that show problems with PHP > 7.1 and MariaDB > 10.2. Archlinux has more recent packages on board. Thus I am asking what you did, this might help to sort out other things.

**Comment by tsichevski** [comment]
Cannot import anything to demo.partkeepr.org today.
The "Preview" tab in the "Import" dialog reads:
The key importBehaviour does not exist for path /!

Same problem in my Debian9 installation of partkeepr 1.4 i created following the instructions at https://wiki.partkeepr.org/wiki/PartKeepr_on_Debian_%22Stretch%22

**Comment by boiddude** [comment]
I'm trying to import from a CSV file. I can choose the file, Set the fields I want to match. I click on Execute import and it brings up a second window with what it seemed to have imported. I just can't find the information it supposedly imported anywhere. Does anyone have any ideas?  I get this error when I click on **Execute import**. 

[2020-07-22 00:36:05] request.CRITICAL: Uncaught PHP Exception Dunglas\ApiBundle\Exception\InvalidArgumentException: "No route matches "undefined"." at /var/www/html/vendor/dunglas/api-bundle/Api/IriConverter.php line 76 {"exception":"[object] (Dunglas\\ApiBundle\\Exception\\InvalidArgumentException(code: 0): No route matches \"undefined\". at /var/www/html/vendor/dunglas/api-bundle/Api/IriConverter.php:76, Symfony\\Component\\Routing\\Exception\\ResourceNotFoundException(code: 0):  at /var/www/html/app/cache/prod/appProdProjectContainerUrlMatcher.php:3087)"} []

**Comment by christianlupus** [comment]
@boiddude what PHP version and what database (incl version) are you using? Just to rule out some incompatibility with the outdated Symfony framework.

**Comment by boiddude** [comment]
My PHP level is 7.3.11 and the database is Mariadb. It is a docker-compose set of images made by mhubig and running on Ubuntu 20.04. I see that there are some prerequisites that have to be fulfilled as far as 2 errors are concerned but I have them supposedly taken care of as the errors go away.  I have to use the same data from my CVS file to take care of them though so it is being imported under different keys in the import.

**Comment by christianlupus** [comment]
If you are referring to [this repositorie's Docker image](https://github.com/mhubig/docker-partkeepr/blob/master/Dockerfile), it is PHP 7.1. In this case I will just keep this issue open for further reference.

**Comment by boiddude** [comment]
Yes, I just was looking at the PHP on my system.  Do you know how to import?  What is the process.  Some people talked about projects.  I see on the main Part Manager page and to bottom middle that there is an import option there as well. I have tried both of them honestly and neither of them actually import. Can you maybe run through the steps?  I only have 4 columns of data in my CSV file.  One is the part number, then the quantity, the description and then the location.

**Comment by boiddude** [comment]
Ok, I was able to get rid of the errors by setting the PartKeepr.PartBundle.Entity.Part name to the part number. I was able to set the category description to the part description so that the errors went away. I am clicking on the Preview tab and it is saying "The part has no category assigned" but doesn't list a part.  What am I supposed to do? It won't import now and was importing yesterday even though it wasn't but it acted like it was.
